lib/msal-core/src/utils/WindowUtils.ts

```diff
@@ -139,7 +139,10 @@ export class WindowUtils {
                 if (href && UrlUtils.urlContainsHash(href)) {
                     logger.verbose("monitorPopupForHash found url in hash");
                     clearInterval(intervalId);
-                    resolve(contentWindow.location.hash);
+                    const hash = contentWindow.location.hash;
+                    contentWindow.location.hash = "";
+                    contentWindow.history.replaceState(null, null, `${contentWindow.location.pathname}${contentWindow.location.search}`);
```

Comment:
As with redirect, you should check if `history.replaceState` is a function first.

Reply:
Seems like this logic could be factored out.